packages/components/src/components/assets/AssetIcon/AssetIcon.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from 'react';
+import {
+    AssetShareIndicator,
+    AssetShareIndicatorProps,
+} from '../AssetShareIndicator/AssetShareIndicator';
+import { useTheme } from 'styled-components';
+
+export const UPDATED_ICONS_LIST_URL_BASE = 'https://data.trezor.io/suite/icons/coins/';
+
+const QUALITY = {
+    small: '',
+    big: '@2x',
```

Comment:
nitpick: what about `normal` instead of `small`. Because we can use also @3x etc.

Reply:
Just a note that later it was decided to use the terms "small" and "medium", 16px and 24px, and for both to use "@2x" suffix in the url to get a nicer good quality image for those sizes, at least for now.

And what about bigger sizes? That's discussion in progress.

Reply:
`standard` will very probably never be used as we've found the smallest possible quality to achieve nice sharp icon look is what you call `double`. I wanted to avoid adding code that would probably never be used. And for `triple` quality, I was told bigger icons than 24px are unfortunately not available, and it doesn't seem they will.

Are you ok with using the following?

const QUALITY = {
    double: '@2x',
}

IMHO unnecessary but it could make some parts of the code more understandable.

Reply:
In this case why do we need quality since we use just `@2x`? Let's delete `quality` prop then and component will be even simpler, WDYT?

Reply:
Not sure. We still need to distinguish between smaller 16px and bigger 24px icon. Originally, this was meant kinda differently but since we've found we need @2x also for the smaller icon..

Reply:
But relation between quality and size should be this:

```
finalResolution = imageSize * quality 
```

So if quality is always `@2x`, we don't need to change it. If we have two sizes of icons (16px and 24px), resolution should be like this:

```
// size = "small"
finalResolution = 16 * 2

// size = "medium"
finalResolution = 24 * 2
```

Reply:
I've decided to use quality as it is because `AssetShareIndicator` already uses `size` and we don't wanna mix with that, while keeping the original behavior of the component. Why to use both? With `quality` we make sure we choose one of the 2 sizes for the icon (16 and 24px) when not using `symbol` but using the generated url as the icon's source.

Reply:
Do we need different sizes then 16/24? Can we unify it? So it doesn't matter if we are using symbol or not.

I suggest exploring the possibility of allowing only 16px and 24px and just leave it like this everywhere. 

Additionally I would use x2 everywhere so 32 and 48. So in components API we have those two resolutions. If we need bigger ones.


And event to add to is we may have 16,24,32,48 resolution available and live with this.

Reply:
Sounds good to me. 

Ideally we should use just one prop for setting size without dependency to setting symbol. Otherwise component is not going to be predictable.

Reply:
Please check the issue ticket or reach @MiroslavProchazka or @Hermez-cz for more discussion about that. The need for two specific sizes was requested while using `quality` prop (first it was 24 and 48px-for quality2x, later 16 and 24 while using 2x for both but we needed to distinguish between those 2 sizes somehow anyway). So I've implemented that. I believe there were many reasons behind what was expected to implement.

> Ideally we should use just one prop for setting size without dependency to setting symbol. Otherwise component is not going to be predictable.

This component works like this: if symbol defined, it will be working as before - using size etc. If props for url defined, quality prop will be taken into account and one of the 2 sizes for the icon will be chosen. With this approach we make sure nothing will be broken for existing usages of the components, just new functionality added. 

Here's another why: we cannot use bigger sizes for icons downloaded via that new generated url because the images are not big enough and it wouldn't look ok. Of course this was already tested and discussed before I made these changes.

Reply:
I asked @MiroslavProchazka and @Hermez-cz and they didn't requested `quality` prop but size of the icon.

Before this PR we used `AssetShareIndicator` only once with `size={24}` so @peter-sanderson suggested that we could provide only predefined sizes (`16 | 24 | 32 | 48`) and in this case we don't have to care about `symbol`.
It will simplify API of the component and you could use it without studying inner logic of the component.

Reply:
I do agree simplification would be great, but I must disagree with the rest, as I was involved in some discussions earlier and have kinda different information, and the thing with sizes and qualities is more complicated. And this discussion is already too long, so to be simple, we must distinguish between sizes provided for symbol vs for the url. We cannot offer all to all. At least not for now. We don't have icons ready for that, unfortunately. So we must go for less simple solution if we ever want that.

> Before this PR we used AssetShareIndicator only once with size={24} ...
That could be ok, but with AssetLogo not from various reasons.